### PR TITLE
SE: Fix S4158 FP: Interfaces are only half tracked

### PR DESCRIPTION
**Pull request by Tim-Pohlmann**
```
IList<int> iList = new List<int>();
iList.Add(5);
iList.Clear();                                      // Noncompliant FP
```

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs [hunk 1]
```diff
@@ -474,6 +474,25 @@ class CustomCollection
         public void AddMethodWithDifferentName(int item) { }
     }
 
+    public void AssignmentToDifferentType()
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs:477]: What about?
```suggestion
    public void AssignmentToInterface()
```

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM - small nitpick + needs some more tests (discussed offline).

**Review by mary-georgiou-sonarsource — COMMENTED** [review]
some more tests.

**Review by mary-georgiou-sonarsource — APPROVED** [review]
LGTM

**Comment by Tim-Pohlmann** [comment]
This is not part of a milestone. The bug was just introduced by #7933

**Pull request merged**